proprietary_plugins/02_pro/smart_upload.md

```diff
@@ -23,20 +23,88 @@ To begin uploading, select _Smart Upload_ in the top right.
+
+<!-- Screenshot of ../../../images/smart_upload/su-unsaved-findings.puml -->
+![PUML Unsaved Findings](../../../images/smart_upload/su-unsaved-findings.png)
+
+Once the product matching is ran, the state of the system will be as follows:
+
+<!-- Screenshot of ../../../images/smart_upload/su-products-matched.puml -->
+![PUML Products Matched](../../../images/smart_upload/su-products-matched.png)
+
+#### What happens when a matching product is not found?
+
+If a findings endpoints do not match any existing endpoints within the system, then the finding will be split into N findings, where N is the number of unique hosts where this finding was discovered:
+
+<!-- Screenshot of ../../../images/smart_upload/su-split-findings.puml -->
+![PUML Split Findings](../../../images/smart_upload/su-split-findings.png)
+
+#### What happens if multiple matches are found?
+
+Multiple endpoint to product matches is the primary purpose of smart upload! In the diagram above, a finding with multiple hosts/IPs will be split into identical findings and assigned to their respective products. If there are any endpoint hosts without a matching product, then they will also be split into separate findings by host.
+
+#### How is the data organized in a given product?
+
+Engagements and Tests created under a product are intended to be CI/CD focused. In most cases, the engagement will be reused. However, supplying updated arguments to the importer such as version, build ID, commit hash, or branch tag will be used to filter for a specific engagement based on those values. 
+
+![Engagements and Tests](../../../images/smart_upload/su-test-engagements.png)
+
+### Product Placement
+
+Findings parsed from a given report that are not matched to an existing product will be placed in a "product placement" queue where each finding can be assessed by there title, severity, endpoints where the vulnerability was detected, and what file the finding came from.
+
+![Product Placement](../../../images/smart_upload/su-placement.png)
+
+After selecting at least one finding, three options will be displayed:
 
 - _Create Product_ - create a new product for the finding to be assigned to
 - _Assign Product_ - list all existing products that the finding may be assigned to
-- _Disregard Findings_ - delete the unassigned finding altogether
+- _Discarded Findings_ - delete the unassigned finding altogether
 
-![Product Placement](../../../images/smart_upload/su-placement.png)
+#### Create Product
+
+Sometimes, a finding is detected on an endpoint that may not fully align with any existing products, or may not even fit product model directly. In these cases, it is most appropriate to create a new product for these findings to avoid having them getting misplaced in the future. After clicking the "Create Product" button, a short form will appear to gather some required information about the product:
```

Comment:
```suggestion
Sometimes, a finding is detected on an endpoint that may not fully align with any existing products, or it may not even fit the product model directly. In these cases, it is most appropriate to create a new product for these findings to avoid having them getting misplaced in the future. After clicking the "Create Product" button, a short form will appear to gather some required information about the product:
```